### PR TITLE
feat: add pino logger support

### PR DESCRIPTION
**Pull request by beeme1mr**
Signed-off-by: Michael Beemer <beeme1mr@users.noreply.github.com>

## This PR

- adds pino logger

### Notes

Setting the environment variable `NODE_ENV` to `production` enables JSON logging.

### CODE REVIEW

File: packages/app/src/app/app.module.ts [hunk 1]
```diff
@@ -33,6 +40,26 @@ OpenFeature.setTransactionContextPropagator(new AsyncLocalStorageTransactionCont
...
+      pinoHttp: {
+        /**
+         * Pretty print logs when in a non-prod context.
+         *
+         * Note: NX uses webpack behind the scenes which replaces NODE_ENV at
+         * build time. The environment defaults to development unless the `--prod`
+         * flag is used provided in the build command.
+         */
+        transport:
+          process.env['NODE' + '_ENV'] !== 'production'
```
[the remark this answers is not in the document]
Comment (beeme1mr) [reply packages/app/src/app/app.module.ts:53]: I'll take a look.

File: packages/app/src/app/provider.service.ts [hunk 2]
```diff
@@ -37,6 +38,7 @@ export class ProviderService {
         } else {
           return new OpenFeatureLaunchDarklyProvider({
             sdkKey,
+            logger: new OpenFeatureLogger(`LaunchDarkly Provider`),
```
Comment (toddbaert) [inline packages/app/src/app/provider.service.ts:41]: I guess you like the idea to add loggers to the constructors?

Comment (beeme1mr) [reply packages/app/src/app/provider.service.ts:41]: It was just a quick way to remove the console logs.

Comment (beeme1mr) [reply packages/app/src/app/provider.service.ts:41]: But it seems to work fine.

File: packages/app/src/app/app.module.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (beeme1mr) [reply packages/app/src/app/app.module.ts:53]: https://stackoverflow.com/questions/58090082/process-env-node-env-always-development-when-building-nestjs-app-with-nrwl-nx

[the remark this answers is not in the document]
Comment (beeme1mr) [reply packages/app/src/app/app.module.ts:53]: I'm sure I could find a more elegant solution but I think this is good enough for the demo app.

**Pull request merged**